nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -92,9 +94,28 @@ private void initNsisExecutable(GenericFactory factory)
 			BinaryReader blockReader = new FactoryBundledWithBinaryReader(factory,
 					blockDataByteProvider, NsisConstants.IS_LITTLE_ENDIAN);
 			this.commonHeader = new NsisCommonHeader(blockReader);
+			this.sections = getSections(blockReader);
 		}
 	}
 
+	/**
+	 * Get an array of the right amount of sections in the Nsis executable. The
+	 * reader object is expected to be at the right offset (at the beginning of the
```

Comment:
Could you please add a comment that this function modifies `reader` and puts the index to the first byte after read sections?
I feel like it would have been nice to make it explicit for all other methods that consume `BinaryReader`, but that ship has sailed :D

Reply:
ohhh I thought simply stating that the function has to be called with the index at the right place was enough. I'll add a comment stating that the index is advanced